core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java

```diff
@@ -268,53 +247,45 @@ private void reply(org.jgroups.blocks.Response response, Object retVal, Replicab
       }
    }
 
-   protected static Message constructMessage(Buffer buf, Address recipient,ResponseMode mode, boolean rsvp,
-                                             DeliverOrder deliverOrder) {
-      Message msg = new Message();
-      msg.setBuffer(buf);
-      encodeDeliverMode(msg, deliverOrder);
+   static RequestOptions constructRequestOptions(ResponseMode mode, boolean rsvp, DeliverOrder deliverOrder,
+                                                 long timeout) {
+      RequestOptions options = new RequestOptions(mode, timeout);
+      encodeDeliverMode(options, deliverOrder);
       //some issues with the new bundler. put back the DONT_BUNDLE flag.
       if (deliverOrder == DeliverOrder.NONE || mode != ResponseMode.GET_NONE) {
-         msg.setFlag(Message.Flag.DONT_BUNDLE.value());
+         options.setFlags(Message.Flag.DONT_BUNDLE.value());
       }
       // Only the commands in total order must be received by the originator.
       if (deliverOrder != DeliverOrder.TOTAL) {
-         msg.setTransientFlag(Message.TransientFlag.DONT_LOOPBACK.value());
+         options.setTransientFlags(Message.TransientFlag.DONT_LOOPBACK.value());
       }
       if (rsvp) {
-         msg.setFlag(Message.Flag.RSVP.value());
+         options.setFlags(Message.Flag.RSVP.value());
       }
-
-      if (recipient != null) msg.setDest(recipient);
-      return msg;
+      return options;
    }
 
-   Buffer marshallCall(Marshaller marshaller, ReplicableCommand command) {
-      Buffer buf;
+   Buffer marshallCall(ReplicableCommand command) {
       try {
-         buf = marshaller.objectToBuffer(command);
+         return new Buffer(ispnMarshaller.objectToByteBuffer(command));
```

Comment:
The `byte[]` we create could spend a lot of time in UNICAST3/NAKACK2's retransmission tables, so I'd rather keep on trimming it when its way too large. TBH I just assumed our `MarshallerAdapter` did the same, I never thought that it might trust the size predictor 100%.